include/wil/registry_basic.h

```diff
@@ -0,0 +1,1540 @@
+//*********************************************************
+//
+//    Copyright (c) Microsoft. All rights reserved.
+//    This code is licensed under the MIT License.
+//    THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF
+//    ANY KIND, EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED
+//    TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A
+//    PARTICULAR PURPOSE AND NONINFRINGEMENT.
+//
+//*********************************************************
+// ReSharper disable CppInconsistentNaming
+#ifndef __WIL_REGISTRY_BASIC_INCLUDED
+#define __WIL_REGISTRY_BASIC_INCLUDED
+
+// TODO: should we use feature checks instead of forced imports?
+#include <functional>
+#include <iterator>
+#include <memory>
+#include <string>
+#include <type_traits>
+#include <utility>
+#include <vector>
+#include <list>
+
+#include <Windows.h>
+#include <sddl.h>
+
+#include "common.h"
+#include "resource.h"
+
+#ifdef _KERNEL_MODE
+#error This header is not supported in kernel-mode.
+#endif
+
+#ifndef __WIL_WINREG_
+#error This required wil::unique_hkey support
+#endif
+
+namespace wil
+{
+namespace registry
+{
+constexpr bool error_value_was_not_found(HRESULT hr) WI_NOEXCEPT

```

Comment:
These seem to be unused

Reply:
yeah. they are on the public wil::registry namespace. i.e. friendly functions if callers don't want to hunt down error codes themselves.